main.go

```diff
@@ -311,3 +321,23 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options)
 
 	return nil
 }
+
+func autoDetectClusterDomain() string {
+	cname, err := net.LookupCNAME(k8sDefaultService)
+	if err != nil {
+		setupLog.Error(err, "unable to autodetect cluster domain")
+		return ""
+	}
+
+	re := regexp.MustCompile(regexp.QuoteMeta(k8sDefaultService) + `\.(.*)\.`)
```

Comment:
Can we split the rest of this function into a parseClusterDomainCName function or something and add some unit tests?

Reply:
Done: https://github.com/spiffe/spire-controller-manager/pull/90/commits/b51634319ff9e5933c300fde85f573cdaa6e2f20